source/Calamari.AzureAppService/Behaviors/TargetDiscoveryBehaviour.cs

```diff
@@ -210,21 +210,35 @@ public static ServiceMessage CreateWebAppTargetCreationServiceMessage(string? re
                 "create-azurewebapptarget",
                 parameters.Where(p => p.Value != null).ToDictionary(p => p.Key, p => p.Value!));
         }
-
-        public static async Task<AzureResource[]> GetResourcesByType(this ArmClient armClient, params string[] types)
+        
+        public static async Task<AzureResource[]> GetResourcesByType(
+            this ArmClient armClient,
+            params string[] types)
         {
             var tenant = armClient.GetTenants().First();
-
-            var typesToRetrieveClause = string.Join(" or ", types.Select(t => $"type == '{t}'"));
             var typeCondition = types.Any()
-                ? $"| where { typesToRetrieveClause } |"
+                ? $"| where {string.Join(" or ", types.Select(t => $"type == '{t}'"))} |"
                 : string.Empty;
 
-            var query = new ResourceQueryContent(
-                $"Resources {typeCondition} project name, type, tags, resourceGroup");
+            var query = new ResourceQueryContent($"Resources {typeCondition} project name, type, tags, resourceGroup")
+            {
+                Options = new ResourceQueryRequestOptions { Top = 1000 }
+            };
+
+            var allResults = new List<AzureResource>();
 
-            var response = await tenant.GetResourcesAsync(query, CancellationToken.None);
-            return JsonConvert.DeserializeObject<AzureResource[]>(response.Value.Data.ToString())!;
+            while (true)
+            {
+                var page = (await tenant.GetResourcesAsync(query, CancellationToken.None)).Value;
+                var batch = JsonConvert.DeserializeObject<AzureResource[]>(page.Data.ToString())!;
+                allResults.AddRange(batch);
+
+                if (string.IsNullOrEmpty(page.SkipToken))
+                    break;
+
+                query.Options.SkipToken = page.SkipToken;
+            }
```

Comment:
I'm sure this will probably just work, but relying on external data as our exit condition is risky.
I would suggest a maximum loop counter
```suggestion
           var attempt = 0
           var maxAttempts = 10
            while (attempt++ < maxAttempts)
            {
                var page = (await tenant.GetResourcesAsync(query, CancellationToken.None)).Value;
                var batch = JsonConvert.DeserializeObject<AzureResource[]>(page.Data.ToString())!;
                allResults.AddRange(batch);

                if (string.IsNullOrEmpty(page.SkipToken))
                    break;

                query.Options.SkipToken = page.SkipToken;
            }
```